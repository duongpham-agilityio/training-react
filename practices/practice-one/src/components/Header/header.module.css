.mainHeader {
  font-family: var(--primary-font);
  box-shadow: 0 0.05mm 0;
}

.mainHeader .mainNavbar {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.mainHeader .headerItem {
  display: flex;
  align-items: center;
}

.mainHeader .headerItem:nth-child(1) {
  gap: 20px;
}

.mainHeader .headerLogo {
  width: 100px;
}

.mainHeader .navLink {
  font-weight: 600;
  color: var(--primary-text-color);
}

.mainHeader .headerFilter {
  display: flex;
  overflow: hidden;
  border: 1px solid var(--primary-color);
  border-radius: 4px;
}

.mainHeader .headerFilter .btn,
.mainHeader .btn.headerCartIcon {
  border: none;
  border-radius: unset;
}

.mainHeader .headerFilter .input {
  border-radius: unset;
  border: none;
  margin: unset;
}

.mainHeader .headerCartIcon {
  position: relative;
  font-size: 17px;
  background-color: transparent;
  padding-left: 15px;
  padding-right: 15px;
}

.mainHeader .headerCartIcon.show::after {
  position: absolute;
  top: -8px;
  right: 0;
  content: attr(data-size);
  border-radius: 50%;
  color: white;
  font-size: 12px;
  font-weight: 600;
  background-color: red;
  padding: 1px 5px;
}
