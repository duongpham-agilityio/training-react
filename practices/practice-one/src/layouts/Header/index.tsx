import { ChangeEvent, memo, MouseEvent } from 'react';
import { Link } from 'react-router-dom';
import { useApp } from 'hooks';
import { FaFilter, FaSearch, FaShoppingBag } from 'react-icons/fa';

// Components
import { Button, Input } from 'components/commons';

// Constants
import { END_POINT } from '@constants';

// Styles
import headerStyles from './header.module.css';
import containerStyles from 'styles/commons/container.module.css';

// Images
import logo from 'assets/images/logo.jpg';

type HeaderProps = {
  isFilter?: boolean;
  // eslint-disable-next-line no-unused-vars
  onOpenCart: (event: boolean) => void;
  // eslint-disable-next-line no-unused-vars
  onOpenFilter: (event: MouseEvent) => void;
  // eslint-disable-next-line no-unused-vars
  onSearch: (event: ChangeEvent) => void;
};

const Header = (props: HeaderProps) => {
  const {
    globalState: { cartSize },
  } = useApp();
  const { isFilter = true, onOpenFilter, onSearch, onOpenCart } = props;

  return (
    <header
      className={`${headerStyles.mainHeader} ${containerStyles.container}`}
    >
      <nav className={headerStyles.mainNavbar}>
        <div className={headerStyles.headerItem}>
          <a href={END_POINT.HOME}>
            <img src={logo} alt="logo" className={headerStyles.headerLogo} />
          </a>
          <ul className={headerStyles.navlists}>
            <li className={headerStyles.navItem}>
              <Link to="/" className={headerStyles.navLink}>
                Home
              </Link>
            </li>
          </ul>
        </div>
        <div className={headerStyles.headerItem}>
          <div className={headerStyles.headerFilter}>
            {isFilter && (
              <Button
                description="Filters"
                rightIcon={<FaFilter />}
                className={headerStyles.btn}
                onClick={onOpenFilter}
              />
            )}
            <Input
              placeholder="Search product..."
              leftIcon={<FaSearch />}
              className={headerStyles.input}
              onChange={onSearch}
            />
          </div>
          {/* add class 'show' when add size cart > 0 */}
          {Boolean(cartSize) && (
            <Button
              className={`${headerStyles.headerCartIcon} ${headerStyles.show}`}
              data-size={cartSize}
              onClick={(e: MouseEvent) => {
                e.stopPropagation();
                onOpenCart(true);
              }}
            >
              <FaShoppingBag />
            </Button>
          )}
        </div>
      </nav>
    </header>
  );
};

export default memo(Header);
