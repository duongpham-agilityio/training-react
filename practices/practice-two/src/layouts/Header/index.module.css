.mainHeader {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 37px;
  box-shadow: 0 0 4px rgba(0, 0, 0, 0.25);
}

.mainHeader .heading {
  font-weight: 400;
}
