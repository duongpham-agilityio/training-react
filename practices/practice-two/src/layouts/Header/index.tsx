import { Link } from 'react-router-dom';

// Components
import { Heading } from 'components/commons';

// Styles
import headerStyles from 'layouts/Header/index.module.css';

// Assets
import Logo from 'assets/images/Logo.jpg';

const Header = () => {
  return (
    <header className={headerStyles.mainHeader}>
      <Link to="/">
        <img src={Logo} alt="logo" width="100px" height="60px" />
      </Link>
      <Heading
        label="Hello! Welcome to the Book Marker"
        className={headerStyles.heading}
      />
    </header>
  );
};

export default Header;
