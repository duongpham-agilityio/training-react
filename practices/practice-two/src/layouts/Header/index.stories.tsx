import { Meta, Story } from '@storybook/react';

import { Header } from 'layouts';
import { BrowserRouter } from 'react-router-dom';

export default {
  title: 'Practice/layouts/Header',
  component: Header,
} as Meta;

export const Index: Story = () => (
  <BrowserRouter>
    <Header />
  </BrowserRouter>
);
