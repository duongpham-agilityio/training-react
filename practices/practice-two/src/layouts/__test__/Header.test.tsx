import { render } from '@testing-library/react';

// Components
import Header from 'layouts/Header';
import { BrowserRouter } from 'react-router-dom';

const setup = () =>
  render(
    <BrowserRouter>
      <Header />
    </BrowserRouter>
  );

describe('Header', () => {
  it('Match to snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
