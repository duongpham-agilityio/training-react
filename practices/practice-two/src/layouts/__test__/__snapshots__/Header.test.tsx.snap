// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Match to snapshot 1`] = `
<div>
  <header>
    <a
      href="/"
    >
      <img
        alt="logo"
        height="60px"
        src="default"
        width="100px"
      />
    </a>
    <h2
      class="undefined undefined "
    >
      Hello! Welcome to the Book Marker
    </h2>
  </header>
</div>
`;
